raw.go

```diff
@@ -34,14 +34,10 @@ func DecodeRawBlock(block blocks.Block) (ipld.Node, error) {
 
 var _ ipld.DecodeBlockFunc = DecodeRawBlock
 
-// NewRawNodeWPrefix creates a RawNode with the hash function
-// specified in prefix.
-func NewRawNodeWPrefix(data []byte, prefix cid.Prefix) (*RawNode, error) {
-	prefix.Codec = cid.Raw
-	if prefix.Version == 0 {
-		prefix.Version = 1
-	}
-	c, err := prefix.Sum(data)
+// NewRawNodeWPrefix creates a RawNode using the provided cid builder
+func NewRawNodeWPrefix(data []byte, builder cid.Builder) (*RawNode, error) {
+	builder = builder.WithCodec(cid.Raw)
```

Comment:
Note: leaving this one alone for now.  I personally think we should change NewRawNode to always take a builder rather than creating a separate function, it can take nil if the user wishes to use the default.